ocaml/sdk-gen/java/templates/Types.mustache

```diff
@@ -0,0 +1,253 @@
+    */
+   public static class XenAPIException extends IOException {
+       public final String shortDescription;
+       public final String[] errorDescription;
+       XenAPIException(String shortDescription)
+       {
+           this.shortDescription = shortDescription;
+           this.errorDescription = null;
+       }
+       XenAPIException(String[] errorDescription)
+       {
+           this.errorDescription = errorDescription;
+           if (errorDescription.length > 0)
+           {
+               shortDescription = errorDescription[0];
+           } else
+           {
+               shortDescription = "";
+           }
+       }
+       public String toString()
+       {
+           if (errorDescription == null)
+           {
+               return shortDescription;
+           } else if (errorDescription.length == 0)
+           {
+               return "";
+           }
+           StringBuilder sb = new StringBuilder();
+           for (int i = 0; i < errorDescription.length - 1; i++)
+           {
+               sb.append(errorDescription[i]);
+           }
+           sb.append(errorDescription[errorDescription.length - 1]);
+           return sb.toString();
+       }
+   }
+
+   /**
+    * Thrown if the response from the server contains an invalid status.
+    */
+   public static class BadServerResponse extends XenAPIException
+   {
+       public BadServerResponse(JsonRpcResponseError responseError)
+       {
+           super(String.valueOf(responseError));
+       }
+   }
+   /**
+   * Checks the provided server response was successful. If the call
+   * failed, throws a XenAPIException. If the server
+   * returned an invalid response, throws a BadServerResponse.
+   * Otherwise, returns the server response as passed in.
+   */
+   public static void checkError(JsonRpcResponseError response) throws XenAPIException, BadServerResponse
+   {
+        var errorData = response.data;
+        if(errorData.length == 0){
+            throw new BadServerResponse(response);
+        }
+        var errorName = response.message;
+
+        {{#errors}}
+        if (errorName.equals("{{{name}}}")){
+            {{#err_params}}
+            String p{{{index}}} = errorData.length > {{{index}}} ? errorData[{{{index}}}] : "";
+            {{/err_params}}
+            throw new Types.{{{class_name}}}({{#err_params}}p{{{index}}}{{^last}}, {{/last}}{{/err_params}});
+        }
+        {{/errors}}
+
+        // An unknown error occurred
+        throw new Types.XenAPIException(errorData);
+    }
+
+    {{#enums}}
+    public enum {{{class_name}}} {
+        /**
+         * The value does not belong to this enumeration
+         */
+        @JsonEnumDefaultValue
+        UNRECOGNIZED,
+        {{#values}}
+        /**
+         * {{{description}}}
+         */
+        @JsonProperty("{{{name}}}")
+        {{{name_uppercase}}}{{^is_last}},{{/is_last}}{{#is_last}};{{/is_last}}
+        {{/values}}
+        public String toString() {
```

Comment:
Not caused by these changes, but can we insert a line above here so it can breathe a bit.